apps/web/src/services/cost-explorer/cost-analysis/modules/CostAnalysisDataTable.vue

```diff
@@ -306,21 +310,18 @@ const handleChange = async (options: any = {}) => {
     if (costAnalysisPageState.period) tableState.items = getRefinedChartTableData(results, costAnalysisPageState.granularity, costAnalysisPageState.period);
     tableState.more = more;
 };
+const costAnalyzeExportApiQueryHelper = new ApiQueryHelper();
 const handleExcelDownload = async () => {
     try {
-        analyzeApiQueryHelper.setFilters(costAnalysisPageStore.consoleFilters);
-        const dateFormat = costAnalysisPageState.granularity === GRANULARITY.MONTHLY ? 'YYYY-MM' : 'YYYY-MM-DD';
-
-
+        costAnalyzeExportApiQueryHelper.setFilters(costAnalysisPageStore.consoleFilters);
         await store.dispatch('file/downloadExcel', {
             url: '/cost-analysis/cost/analyze',
             param: {
-                granularity: costAnalysisPageState.granularity,
-                group_by: costAnalysisPageState.groupBy,
-                start: dayjs.utc(costAnalysisPageState?.period?.start).format(dateFormat),
-                end: dayjs.utc(costAnalysisPageState?.period?.end).format(dateFormat),
-                filter: analyzeApiQueryHelper.data.filter,
-                query: analyzeApiQueryHelper.data,
+                data_source_id: costAnalysisPageStore.selectedDataSourceId,
+                query: {
+                    ...state.analyzeQuery,
+                    ...costAnalyzeExportApiQueryHelper.data,
```

Comment:
nitpick: Please check if `costAnalyzeExportApiQueryHelper.data` doesn't override or add anything like sort or pagination.

Reply:
I checked it!